docs/howto/danceable-lyrics-input-tool/README.md

```diff
@@ -0,0 +1,485 @@
+        .select(
+            pl.col("title").str.to_lowercase().alias("track_name"),
+            pl.col("artist").str.to_lowercase().alias("artist_name"),
+            "lyrics",
+            "language",
+            "views",
+        )
+        .filter(
+            pl.col("lyrics").str.contains("(?i)" + "|(?i)".join(self.LYRICS_TERMS))
+        )
+        .filter(pl.col("language") == "en")
+        .filter(pl.col("views") > self.MIN_VIEWS)
+        .collect()
+    )
+```
+
+Above: We are telling Polars to create a `DataFrame` containing select columns from rows matching our criteria. We are also normalizing some column names by lowercasing them and aliasing them in our output. In our `filter` clause, we are building a case-insensitive [Regular Expression](https://en.wikipedia.org/wiki/Regular_expression) from our list of terms. Finally, we filter out non-English terms and restrict to only entries with more than `self.MIN_VIEWS`. You may wish to alter this criteria a bit, for example, by allowing other languages.
+
+The next section of code to build will read from multiple input files and join them into a single `danceable_tracks` view of the data containing columns restricted to our criteria.
+
+```python
+self.provider.io.info(f"{self.name} gathering danceable track information...")
+
+artists = (
+    pl.scan_csv(self.DATASETS_BASE + "artists.csv")
+    .select(
+        pl.col("name").str.to_lowercase().alias("artist_name"),
+        pl.col("id").alias("artist_id"),
+    )
+    .collect()
+)
+
+tracks = (
+    pl.scan_csv(self.DATASETS_BASE + "tracks.csv")
+    .select(
+        pl.col("name").str.to_lowercase().alias("track_name"),
+        pl.col("id").alias("track_id"),
+        "explicit",
+    )
+    .filter(pl.col("explicit") == 0)
+    .collect()
+)
+
+track_artists = (
+    pl.scan_csv(self.DATASETS_BASE + "r_track_artist.csv")
+    .select(pl.col("track_id"), pl.col("artist_id"))
+    .sort("track_id")
+    .collect()
+)
+
+audio_features = (
+    pl.scan_csv(self.DATASETS_BASE + "audio_features.csv")
+    .select(pl.col("id").alias("track_id"), "danceability", "energy", "tempo",)
+    .filter(pl.col("danceability").is_between(*self.DANCEABILITY_RANGE))
+    .filter(pl.col("energy").is_between(*self.ENERGY_RANGE))
+    .filter(pl.col("tempo").is_between(*self.TEMPO_RANGE))
+    .sort("track_id")
+    .collect()
+)
+
+danceability_tracks = (
+    track_artists.join(artists, on="artist_id")
+    .join(tracks, on="track_id")
+    .join(audio_features, on="track_id")
+)
+```
+
+If we were to `print(danceability_tracks)` we'd see a header similar to the following:
+
+```txt
+track_id | artist_id | artist_name | track_name | explicit | danceability | energy | tempo
+```
+As you can see, this is a combination (joined) from multiple sources.
+
+
+Now that we have both a sample of songs, their lyrics containing our terms, and track information including "danceability", let's find our matches! Add the following code:
```

Comment:
TW review:

Now that we have both a sample of songs with lyrics that contain our terms and track information that includes "danceability", let's find our matches! Add this code: